client/my-sites/customer-home/cards/features/domain-upsell/index.jsx

```diff
@@ -23,19 +23,24 @@ export default function DomainUpsell() {
 	const site = useSelector( ( state ) => getSelectedSite( state ) );
 	const isEmailVerified = useSelector( ( state ) => isCurrentUserEmailVerified( state ) );
 	const siteDomains = useSelector( ( state ) => getDomainsBySiteId( state, site.ID ) );
+	const isMonthlyOrFreePlan = isMonthly( site.plan?.product_slug );
 
-	if ( siteDomains.length > 1 || ! isEmailVerified || ! isFreePlanProduct( site.plan ) ) {
+	if (
+		siteDomains.filter( ( domain ) => ! domain.isWPCOMDomain ).length ||
+		! isEmailVerified ||
+		( ! isFreePlanProduct( site.plan ) && isMonthlyOrFreePlan )
```

Comment:
Would a function with a documentation describe this logic better?

Reply:
Could be or just an inline comment because it will be smaller when merging this and monthly plans